podpac/core/coordinates/stacked_coordinates.py

```diff
@@ -260,16 +289,39 @@ def __setitem__(self, dim, c):
         if c.name is None:
             c.name = dim
 
-        idx = self.dims.index(dim)    # find the tuple index of the dimension you are interested in
-        coords = list(self._coords)   # make list from _coords tuple
-        coords[idx] = c               # set the element of the coords list to new coordinates
+        d = self._coords.copy()
+        d[dim] = c
 
         # check consistency
-        self._check_sizes([c.size for c in coords])
-        self._check_names([c.name for c in coords])
-        self._check_coord_ref_sys(coords)
+        self._check_sizes([c.size for c in d])
+        self._check_names([c.name for c in d])
+        self._check_coord_ref_sys(d)
+
+        self._coords = d
+
+
+    def __eq__(self, other):
+        if not isinstance(other, StackedCoordinates):
+            return False
+
+        # shortcuts
+        if self.dims != other.dims:
+            return False
+
+        if self.size != other.size:
+            return False
+
+        # full check of underlying coordinates
+        if self._coords != other._coords:
+            return False
+
+        return True
+
+    def __delitem__(self, dim):
+        if not dim in self.dims:
+            raise KeyError("Cannot delete dimension '%s' in StackedCoordinates %s" % (dim, self.dims))
 
-        self.set_trait('_coords', coords)
+        del self._coords[dim]
```

Comment:
I think you need to check that this doesn't leave the StackedCoordinates with only one dimension.